docker-compose/cassandra/custom-cassandra.yaml

```diff
@@ -0,0 +1,1848 @@
+
```

Comment:
please indicate in a comment where the majority of this file came from

Reply:
Hi @yurishkuro  .
I have done adding the custom cassandra config and mount it using volumes in the docker compose , but what I realized when I tried different test, is that we need to add custom cassandra config for each version of cassandra and Jaeger which means we need 4 custom cassandra config .

we need that because mounting a config file needs to specify all the necessary  configs not just those we want to override or change .
the content of the file is just simply a copy of  the running cassandra container with change of th authenticator to PasswordAuthenticator

Reply:
Can you elaborate why 4 different configs?

We already have v3 and v4 folders for docker-compose files, you can co-locate configs there.

Reply:
sorry, but I don't follow. Regardless of HOW we run the test, there are only two versions of Cassandra that we stand up today via v3/v4 docker-compose files, without any extra configuration (meaning it's using its default configuration). The only thing we need to add is authentication. That means we need 2 config files for v3/v4 directory each.

Reply:
Hi  @yurishkuro ,

I found another way to config the cassandra.yaml:

>     command: >
      /bin/sh -c "echo 'authenticator: PasswordAuthenticator' >> /etc/cassandra/cassandra.yaml && docker-entrypoint.sh cassandra -f"

we can add this command in the docker-compose config. it really edit the authenticator to PasswordAuthenticator.
so we don't need to use custom cassandra configs anymore.

Reply:
much better!

Reply:
After I edit the authenticator to PasswordAuthenticator either manually or by the command mentioned above , a puzzling error happens when I run tests for Jaeger v2:

 ```  scripts/cassandra-integration-test.sh 4 v004 v2
+ main 4 v004 v2
+ check_arg 4 v004 v2
+ '[' '!' 3 -eq 3 ']'
+ echo 'Executing integration test for 4 with schema v004.cql.tmpl'
Executing integration test for 4 with schema v004.cql.tmpl
+ run_integration_test 4 v004 v2
+ local version=4
+ local major_version=4
+ local schema_version=v004
+ local jaegerVersion=v2
+ local username=cassandra
+ local password=cassandra
+ local primaryKeyspace=jaeger_v1_dc1
+ local archiveKeyspace=jaeger_v1_dc1_archive
+ local compose_file=docker-compose/cassandra/v4/docker-compose.yaml
+ setup_cassandra docker-compose/cassandra/v4/docker-compose.yaml
+ local compose_file=docker-compose/cassandra/v4/docker-compose.yaml
+ docker compose -f docker-compose/cassandra/v4/docker-compose.yaml up -d
WARN[0000] /home/ali/projects/open-source/jaeger/docker-compose/cassandra/v4/docker-compose.yaml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
[+] Running 2/2
 ✔ Network v4_cassandra-net  Created                                                                                            0.1s 
 ✔ Container v4-cassandra-1  Started                                                                                            2.0s 
+ echo docker_compose_file=docker-compose/cassandra/v4/docker-compose.yaml
+ apply_schema v004 jaeger_v1_dc1 cassandra cassandra
+ local image=cassandra-schema
+ local schema_dir=plugin/storage/cassandra/
+ local schema_version=v004
+ local keyspace=jaeger_v1_dc1
+ local username=cassandra
+ local password=cassandra
+ params=('--rm' '--env' 'CQLSH_HOST=localhost' '--env' 'CQLSH_PORT=9042' '--env' 'TEMPLATE=/cassandra-schema/v004.cql.tmpl' '--env' 'KEYSPACE=jaeger_v1_dc1' '--env' 'CASSANDRA_USERNAME=cassandra' '--env' 'CASSANDRA_PASSWORD=cassandra' '--network' 'host')
+ local params
+ docker build -t cassandra-schema plugin/storage/cassandra/
[+] Building 2.6s (8/8) FINISHED                                                                                      docker:default
 => [internal] load build definition from Dockerfile                                                                            0.0s
 => => transferring dockerfile: 349B                                                                                            0.0s
 => [internal] load metadata for docker.io/library/cassandra:5.0                                                                1.9s
 => [internal] load .dockerignore                                                                                               0.0s
 => => transferring context: 2B                                                                                                 0.0s
 => [internal] load build context                                                                                               0.0s
 => => transferring context: 492B                                                                                               0.0s
 => [1/3] FROM docker.io/library/cassandra:5.0@sha256:3de6b2c6fced3a0151f7f4b77a548da82ff907d92c80bc871eb5d262b4baec76          0.0s
 => CACHED [2/3] COPY schema/* /cassandra-schema/                                                                               0.0s
 => CACHED [3/3] RUN groupadd -g 65532 nonroot &&     useradd -u 65532 -g nonroot nonroot --create-home                         0.0s
 => exporting to image                                                                                                          0.1s
 => => exporting layers                                                                                                         0.0s
 => => writing image sha256:1024f89958e095e8f167c7c3301ae73f99b81c386fb65917b828f231228c44f7                                    0.0s
 => => naming to docker.io/library/cassandra-schema                                                                             0.0s
+ docker run --rm --env CQLSH_HOST=localhost --env CQLSH_PORT=9042 --env TEMPLATE=/cassandra-schema/v004.cql.tmpl --env KEYSPACE=jaeger_v1_dc1 --env CASSANDRA_USERNAME=cassandra --env CASSANDRA_PASSWORD=cassandra --network host cassandra-schema
Checking if Cassandra is up at localhost:9042.

Warning: Using a password on the command line interface can be insecure.
Recommendation: use the credentials file to securely provide the password.

WARNING: cqlsh was built against 5.0-rc1, but this server is 4.1.5.  All features may not work!

system       system_distributed  system_traces  system_virtual_schema
system_auth  system_schema       system_views 

Cassandra connection established.
Generating the schema for the keyspace jaeger_v1_dc1 and datacenter dc1.
Using template file /cassandra-schema/v004.cql.tmpl with parameters:
    mode = test
    datacenter = dc1
    keyspace = jaeger_v1_dc1
    replication = {'class': 'SimpleStrategy', 'replication_factor': '3'}
    trace_ttl = 172800
    dependencies_ttl = 0
    compaction_window_size = 96
    compaction_window_unit = MINUTES

Warning: Using a password on the command line interface can be insecure.
Recommendation: use the credentials file to securely provide the password.

WARNING: cqlsh was built against 5.0-rc1, but this server is 4.1.5.  All features may not work!

Warnings :
Your replication factor 3 for keyspace jaeger_v1_dc1 is higher than the number of nodes 1

Schema generated.
+ apply_schema v004 jaeger_v1_dc1_archive cassandra cassandra
+ local image=cassandra-schema
+ local schema_dir=plugin/storage/cassandra/
+ local schema_version=v004
+ local keyspace=jaeger_v1_dc1_archive
+ local username=cassandra
+ local password=cassandra
+ params=('--rm' '--env' 'CQLSH_HOST=localhost' '--env' 'CQLSH_PORT=9042' '--env' 'TEMPLATE=/cassandra-schema/v004.cql.tmpl' '--env' 'KEYSPACE=jaeger_v1_dc1_archive' '--env' 'CASSANDRA_USERNAME=cassandra' '--env' 'CASSANDRA_PASSWORD=cassandra' '--network' 'host')
+ local params
+ docker build -t cassandra-schema plugin/storage/cassandra/
[+] Building 1.7s (8/8) FINISHED                                                                                      docker:default
 => [internal] load build definition from Dockerfile                                                                            0.0s
 => => transferring dockerfile: 349B                                                                                            0.0s
 => [internal] load metadata for docker.io/library/cassandra:5.0                                                                1.1s
 => [internal] load .dockerignore                                                                                               0.0s
 => => transferring context: 2B                                                                                                 0.0s
 => [internal] load build context                                                                                               0.0s
 => => transferring context: 492B                                                                                               0.0s
 => [1/3] FROM docker.io/library/cassandra:5.0@sha256:3de6b2c6fced3a0151f7f4b77a548da82ff907d92c80bc871eb5d262b4baec76          0.0s
 => CACHED [2/3] COPY schema/* /cassandra-schema/                                                                               0.0s
 => CACHED [3/3] RUN groupadd -g 65532 nonroot &&     useradd -u 65532 -g nonroot nonroot --create-home                         0.0s
 => exporting to image                                                                                                          0.1s
 => => exporting layers                                                                                                         0.0s
 => => writing image sha256:1024f89958e095e8f167c7c3301ae73f99b81c386fb65917b828f231228c44f7                                    0.0s
 => => naming to docker.io/library/cassandra-schema                                                                             0.0s
+ docker run --rm --env CQLSH_HOST=localhost --env CQLSH_PORT=9042 --env TEMPLATE=/cassandra-schema/v004.cql.tmpl --env KEYSPACE=jaeger_v1_dc1_archive --env CASSANDRA_USERNAME=cassandra --env CASSANDRA_PASSWORD=cassandra --network host cassandra-schema
Checking if Cassandra is up at localhost:9042.

Warning: Using a password on the command line interface can be insecure.
Recommendation: use the credentials file to securely provide the password.

WARNING: cqlsh was built against 5.0-rc1, but this server is 4.1.5.  All features may not work!

jaeger_v1_dc1  system_auth         system_schema  system_views         
system         system_distributed  system_traces  system_virtual_schema

Cassandra connection established.
Generating the schema for the keyspace jaeger_v1_dc1_archive and datacenter dc1.
Using template file /cassandra-schema/v004.cql.tmpl with parameters:
    mode = test
    datacenter = dc1
    keyspace = jaeger_v1_dc1_archive
    replication = {'class': 'SimpleStrategy', 'replication_factor': '3'}
    trace_ttl = 172800
    dependencies_ttl = 0
    compaction_window_size = 96
    compaction_window_unit = MINUTES

Warning: Using a password on the command line interface can be insecure.
Recommendation: use the credentials file to securely provide the password.

WARNING: cqlsh was built against 5.0-rc1, but this server is 4.1.5.  All features may not work!

Warnings :
Your replication factor 3 for keyspace jaeger_v1_dc1_archive is higher than the number of nodes 1

Schema generated.
+ '[' v2 = v1 ']'
+ '[' v2 == v2 ']'
+ STORAGE=cassandra
+ make jaeger-v2-storage-integration-test
(cd cmd/jaeger/ && go build .)
# Expire tests results for jaeger storage integration tests since the environment might change
# even though the code remains the same.
go clean -testcache
bash -c "set -e; set -o pipefail; go test -race -v -coverpkg=./... -coverprofile cover.out ./cmd/jaeger/internal/integration | sed 's/PASS/✅ PASS/g' | sed 's/FAIL/❌ FAIL/g' | sed 's/SKIP/🔕 SKIP/g'"
=== RUN   TestBadgerStorage
    integration.go:98: This test requires environment variable STORAGE=badger
--- 🔕 SKIP: TestBadgerStorage (0.00s)
=== RUN   TestCassandraStorage
    e2e_integration.go:61: Starting Jaeger-v2 in the background with config file /tmp/TestCassandraStorage3593617967/001/storageCleaner_config.yaml
    e2e_integration.go:69: Writing the Jaeger-v2 output logs into /tmp/TestCassandraStorage3593617967/002/jaeger_output_logs.txt
    e2e_integration.go:77: Writing the Jaeger-v2 error logs into /tmp/TestCassandraStorage3593617967/003/jaeger_error_logs.txt
    e2e_integration.go:89: Running command: [jaeger --config /tmp/TestCassandraStorage3593617967/001/storageCleaner_config.yaml]
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:98: Checking if Jaeger-v2 is available on http://localhost:16686/
    e2e_integration.go:108: HTTP request failed: Get "http://localhost:16686/": dial tcp 127.0.0.1:16686: connect: connection refused
    e2e_integration.go:97: 
                Error Trace:    /home/ali/projects/open-source/jaeger/cmd/jaeger/internal/integration/e2e_integration.go:97
                                                        /home/ali/projects/open-source/jaeger/cmd/jaeger/internal/integration/cassandra_test.go:24
                Error:          Condition never satisfied
                Test:           TestCassandraStorage
                Messages:       Jaeger-v2 did not start
--- ❌ FAIL: TestCassandraStorage (60.01s)
=== RUN   TestCreateStorageCleanerConfig
--- ✅ PASS: TestCreateStorageCleanerConfig (0.03s)
=== RUN   TestElasticsearchStorage
    integration.go:98: This test requires environment variable STORAGE=elasticsearch
--- 🔕 SKIP: TestElasticsearchStorage (0.01s)
=== RUN   TestGRPCStorage
    integration.go:98: This test requires environment variable STORAGE=grpc
--- 🔕 SKIP: TestGRPCStorage (0.00s)
=== RUN   TestKafkaStorage
    integration.go:98: This test requires environment variable STORAGE=kafka
--- 🔕 SKIP: TestKafkaStorage (0.00s)
=== RUN   TestMemoryStorage
    integration.go:98: This test requires environment variable STORAGE=memory_v2
--- 🔕 SKIP: TestMemoryStorage (0.00s)
=== RUN   TestOpenSearchStorage
    integration.go:98: This test requires environment variable STORAGE=opensearch
--- 🔕 SKIP: TestOpenSearchStorage (0.00s)
❌ FAIL
coverage: 0.8% of statements in ./...
❌ FAIL github.com/jaegertracing/jaeger/cmd/jaeger/internal/integration 60.459s
❌ FAIL
make: *** [Makefile:146: jaeger-v2-storage-integration-test] Error 1
+ exit_status=2
+ trap 'teardown_cassandra docker-compose/cassandra/v4/docker-compose.yaml' EXIT
+ teardown_cassandra docker-compose/cassandra/v4/docker-compose.yaml
+ local compose_file=docker-compose/cassandra/v4/docker-compose.yaml
+ docker compose -f docker-compose/cassandra/v4/docker-compose.yaml down
WARN[0000] /home/ali/projects/open-source/jaeger/docker-compose/cassandra/v4/docker-compose.yaml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
[+] Running 2/2
 ✔ Container v4-cassandra-1  Removed                                                                                           10.8s 
 ✔ Network v4_cassandra-net  Removed                                                                                            0.5s 
+ exit 2
```

Jaeger-v2 is not starting. What would be the reason?

Reply:
> After I edit the authenticator to PasswordAuthenticator either manually or by the command mentioned above , a puzzling error happens when I run tests for Jaeger v2:

Hi @yurishkuro ,

This error is ambiguous
would you please help!

Reply:
I suspect it's because you are not passing credentials for Cassandra in the configuration from
* plugin/storage/integration/cassandra_test.go
* and cmd/jaeger/config-cassandra.yaml